torch/__init__.py

```diff
@@ -60,18 +61,41 @@ def _load_global_deps():
     here = os.path.abspath(__file__)
     lib_path = os.path.join(os.path.dirname(here), 'lib', lib_name)
 
-    # NB: If the library doesn't exist, don't fail.  In some settings
-    # (especially in fbcode) there may not be this library, but it doesn't
-    # matter because in those settings you're expected to have setup the
-    # dynamic dependencies properly to start with
-    if os.path.exists(lib_path):
-        ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
-
-
-# See Note [Global dependencies]
-_load_global_deps()
-
-from torch._C import *
+    ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
+
+
+if USE_RTLD_GLOBAL_WITH_LIBTORCH or os.getenv('TORCH_USE_RTLD_GLOBAL'):
+    # Do it the hard way.  You might want to load libtorch with RTLD_GLOBAL in a
+    # few circumstances:
+    #
+    #   1. You're in a build environment (e.g., fbcode) where
+    #      libtorch_global_deps is not available, but you still need
+    #      to get mkl to link in with RTLD_GLOBAL or it will just
+    #      not work.
+    #
+    #   2. You're trying to run PyTorch under UBSAN and you need
+    #      to ensure that only one copy of libtorch is loaded, so
+    #      vptr checks work properly
+    #
+    # If you're using this setting, you must verify that all the libraries
+    # you load consistently use the same libstdc++, or you may have
+    # mysterious segfaults.
+    #
+    if platform.system() != 'Windows':
```

Comment:
Maybe you need to move this if condition to L84, otherwise if `TORCH_USE_RTLD_GLOBAL` is accidentally defined in a Windows environment, it will not work.